### DISCUSSION TITLE
How to isolate `limiter` config when working with horizontal scaling

### DISCUSSION THREAD

**Discussion by michielswaanen** [OptimalBits/bull#2485] [Q&A] [answered]
## Description
Hey, first of all great library. It's really saving me a lot of time! 😄 

I have a question about how to use Bull with my application that supports horizontal scaling.
Currently my queue initialization code looks like this:

```node
BullModule.registerQueue({
  name: "assets",
  limiter: { 
    // Max 1 jobs every 10  seconds
    max: 1,
    duration: 10000, 
  },
})
```

When my application has spun up multiple instance (the horizontal scaler) still only processes 1 job every 10 seconds.
I expected that this `limiter` config wouldn't be shared among the application instances.

## Question
Is it possible to make the `limiter` isolated for each application that connects with Bull/Redis?

Should I look at it differently and create a separate `Queue` per instance?

**Comment by manast** [comment, marked as answer]
Bull is a distributed queue system, therefore the limiters are per queue, independent of how many processes you spawn to process the jobs in a given queue. If you need different limiters then you should use a different queue. Also, if you are new to Bull I recommend you to use BullMQ as the rate-limiter there is more robust than the one in Bull: https://docs.bullmq.io/ https://github.com/taskforcesh/bullmq

**Reply by michielswaanen** [reply]
Hey @manast, thanks for getting back to me. I'm currently using the `groupKey` to identify different server instances. This seems to work, however, I will take a look at BullMQ!

**Reply by manast** [reply]
groupKey is not available in BullMQ anymore. But there are more powerful group related functionalities in the Pro version: https://docs.bullmq.io/bullmq-pro/introduction

**Reply by michielswaanen** [reply]
Thanks I will take look!

**Comment by yanyiting** [comment]
收到，谢谢！
